api/src/main/java/io/opencensus/metrics/LongGaugeMetric.java

```diff
@@ -0,0 +1,151 @@
+ *   DataPoint callbackQueueDataPoint = jobsInQueue.addDataPoint(callbackQueue);
+ *
+ *   void processInboundRequest() {
+ *      inboundQueueDataPoint.inc();
+ *      // Your code here.
+ *      inboundQueueDataPoint.dec();
+ *   }
+ *
+ *   void processCallbackRequest() {
+ *      callbackQueueDataPoint.inc();
+ *      // Your code here.
+ *      callbackQueueDataPoint.dec();
+ *   }
+ * }
+ * }</pre>
+ *
+ * @since 0.17
+ */
+public abstract class LongGaugeMetric {
+
+  /**
+   * Add and returns new Data Point. This is more convenient form when you can want to manually
+   * increase and decrease values as per your service requirements. Must be passed the same number
+   * of label values are were passed to {@link MetricRegistry#addLongGaugeMetric} label keys.
+   *
+   * @param labelValues the list of label values.
+   * @return DataPoint the value of single gauge
+   */
+  public abstract DataPoint addDataPoint(List<LabelValue> labelValues);
+
+  /**
+   * Add and returns new Data Point that reports the value of the object after the function. This is
+   * a self sufficient gauge, slightly more common form of gauge is one that monitors some
+   * non-numeric object. The last argument establishes the function that is used to determine the
+   * value of the gauge when the gauge is collected. Must be passed the same number of label values
+   * are were passed to {@link MetricRegistry#addLongGaugeMetric} label keys.
+   *
+   * @param labelValues the list of label values.
+   * @param obj The state object from which the function derives a measurement.
+   * @param function the function to be called.
+   * @param <T> The type of the object upon which the function derives a measurement.
+   */
+  public abstract <T> void addDataPoint(
+      List<LabelValue> labelValues, @Nullable T obj, ToLongFunction<T> function);
+
+  /**
+   * Returns a Data Point for a gauge without labels.
+   *
+   * @return DataPoint the value of default gauge
+   */
+  public abstract DataPoint getDefaultDataPoint();
+
+  /** The value of a single Gauge. */
+  // TODO(mayurkale) : Rename {DataPoint} to some meaningful name, if needed
+  public abstract static class DataPoint {
+
+    /** Increment the gauge value by one. */
+    public abstract void inc();
+
+    /**
+     * Increment the gauge by the given amount.
+     *
+     * @param amt Amount to add to the gauge.
+     */
+    public abstract void inc(long amt);
+
+    /** Decrement the gauge value by one. */
+    public abstract void dec();
+
+    /**
+     * Decrement the gauge by the given amount.
+     *
+     * @param amt Amount to subtract to the gauge.
+     */
+    public abstract void dec(long amt);
+
+    /**
+     * Set the gauge to the given value.
+     *
+     * @param val to assign to the gauge.
+     */
+    public abstract void set(long val);
+
+    /**
+     * Get the value of the gauge.
+     *
+     * @return The current gauge value.
```

Comment:
for return statements start with lower-case -> "the"

Reply:
Done